cxx4opencl/diff2openclc.txt

```diff
@@ -0,0 +1,120 @@
+=== Difference to OpenCL C
+
+{cpp} for OpenCL targets backwards compatibility to OpenCL C
+in the majority of features. However, it is a different
+language that is derived from {cpp} and therefore it inherits
+its fundamental design principles. Hence {cpp} for OpenCL will
+deviate from OpenCL C in the same areas where {cpp} deviates
+from C.
+
+In this section some of the most common areas where it is
+expected to behave differently are described.
+
+==== Implicit Conversions
+
+{cpp} is much more strict about conversions between types
+especially those that are performed implicitly by the compiler.
+For example it is not possible to convert non-`const` object
+to `const` explicitly. For details please refer to {cpp}17
+(`ISO/IEC 14882:2017`) `[conv]`.
+
+[source,c]
+----------
+void foo(){
+  const int* ptrconst;
+  int* ptr = ptrconst; // rejected in C++ for OpenCL
+}
+----------
+
+The same situation applies to narrowing conversions in
+initialization lists ({cpp}17 `[dcl.init.list]`). 
+
+[source,c]
+----------
+struct mytype {
+ int i;
+};
+void foo(uint par){
+  mytype var = {
+    .i = par
+  };
+}
+----------
+
+Note that some compilers will allow to silence this error
+using some flags (e. g. in Clang `-Wno-error=c++11-narrowing`
+can be used).
+
+Among other common conversions that won't be compiled in C++
+mode there are pointer to integers or integer to pointer
+conversions.
+
+[source,c]
+----------
+void foo(){
+  int* ptr;
+  int i = ptr;
+}
+----------
+
+==== Null Literal
+
+In C and OpenCL C null literal is being defined using other
+language features as it is not represented explicitly i.e.
+commonly it is defined as
+
+[source,c]
+----------
+#define NULL (void*)0
+----------
+
+In {cpp} there is an explicit builtin literal `nullptr`
+that should be used instead ({cpp}17 `[lex.nullptr]`).
+
+{cpp} for OpenCL does not define `NULL` and therefore any
+source code using it must be modified to use `nullptr`
+instead. However as a workaround to avoid large modifications
+`NULL` can also be defined/aliased to `nullptr` in custom
+headers or using command line flag `-D`. It is not recommended
+to reuse C definition of `NULL` in {cpp} for OpenCL as it may
+cause compilation failures in cases that work for C.
+
+[source,c]
+----------
+void foo(){
+ int* ptr = NULL; // inivalid initialization of int* with void*
+}
+----------
+
+==== Use of restrict
+
+C++17 does not support `restrict` and therefore {cpp} for OpenCL
```

Comment:
I'm not familiar with the markup language used in this file, so maybe it's all fine. I've noticed you use `{cpp}` everywhere but here you use `C++17`. Should this be replaced by `{cpp}17` or something?

```suggestion
{cpp}17 does not support `restrict` and therefore {cpp} for OpenCL
```